src/Dapr.Common/Exceptions/DaprExtendedErrorDetail.cs

```diff
@@ -0,0 +1,157 @@
+﻿// ------------------------------------------------------------------------
+// Copyright 2024 The Dapr Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ------------------------------------------------------------------------
+
+namespace Dapr.Common.Exceptions
+{
+    /// <summary>
+    /// Abstract base class of the Dapr extended error detail.
+    /// </summary>
+    public abstract record DaprExtendedErrorDetail(DaprExtendedErrorType ErrorType);
+
+    /// <summary>
+    /// Detail when the type url is unrecognized.
+    /// </summary>
+    /// <param name="TypeUrl">The unrecognized type url.</param>
+    public sealed record DaprUnknownDetail(string TypeUrl) : DaprExtendedErrorDetail(DaprExtendedErrorType.Unknown);
```

Comment:
the original proposal just returned null; given the dapr runtime does seem to return a UnknownDetailType if there is a [cast failure](https://github.com/dapr/kit/blob/main/errors/errors.go#L345); it makes sense to me to try and capture this and provide as much detail to the user as possible.